[BUG] Partcoolingfan not Controllable / MPC Autotuning not Compailable

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

M106 is an Unknown Command
Partcooling starts with Heating the Heater
  #define MPC_INCLUDE_FAN  result in "MPC_INCLUDE_FAN requires at least one fan."
  remove   //#define MPC_INCLUDE_FAN  and leave  #define MPC_AUTOTUNE  result in "Marlin\src\module\temperature.cpp: In member function 'void Temperature::MPC_autotune(uint8_t)':
Marlin\src\module\temperature.cpp:1088:9: error: 'MPC_t {aka struct MPC}' has no member named 'fan255_adjustment'; did you mean 'applyFanAdjustment'?
     mpc.fan255_adjustment = 0.0f;
         ^~~~~~~~~~~~~~~~~
         applyFanAdjustment
*** [.pio\build\mega2560\src\src\module\temperature.cpp.o] Error 1"

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/11225630/Configuration.zip)


### Bug Timeline

_No response_

### Expected behavior

_No response_

### Actual behavior

_No response_

### Steps to Reproduce

_No response_

### Version of Marlin Firmware

Bugfix 2.1.x

### Printer model

Costum

### Electronics

MKS GEN L

### Add-ons

_No response_

### Bed Leveling

ABL Bilinear mesh

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

_No response_

--- a/Marlin/src/lcd/menu/menu_advanced.cpp
+++ b/Marlin/src/lcd/menu/menu_advanced.cpp
@@ -370,6 +370,7 @@ void menu_backlash();
     #if ENABLED(MPC_EDIT_MENU)
 
       #define _MPC_EDIT_ITEMS(N) \
+        MPC_t &mpc = thermalManager.temp_hotend[MenuItemBase::itemIndex].mpc; \
         EDIT_ITEM_FAST_N(float31sign, N, MSG_MPC_POWER_E, &mpc.heater_power, 1, 200); \
         EDIT_ITEM_FAST_N(float31sign, N, MSG_MPC_BLOCK_HEAT_CAPACITY_E, &mpc.block_heat_capacity, 0, 40); \
         EDIT_ITEM_FAST_N(float43, N, MSG_SENSOR_RESPONSIVENESS_E, &mpc.sensor_responsiveness, 0, 1); \
@@ -377,7 +378,6 @@ void menu_backlash();
 
       #if ENABLED(MPC_INCLUDE_FAN)
         #define MPC_EDIT_ITEMS(N) \
-          MPC_t &mpc = thermalManager.temp_hotend[MenuItemBase::itemIndex].mpc; \
           _MPC_EDIT_ITEMS(N); \
           EDIT_ITEM_FAST_N(float43, N, MSG_MPC_AMBIENT_XFER_COEFF_FAN_E, &editable.decimal, 0, 1, []{ \
             thermalManager.temp_hotend[MenuItemBase::itemIndex].applyFanAdjustment(editable.decimal); \

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -1085,7 +1085,9 @@ volatile bool Temperature::raw_temps_ready = false;
           block_responsiveness = -log((t2 - asymp_temp) / (t1 - asymp_temp)) / (sample_distance * (sample_count >> 1));
 
     mpc.ambient_xfer_coeff_fan0 = mpc.heater_power * (MPC_MAX) / 255 / (asymp_temp - ambient_temp);
-    mpc.fan255_adjustment = 0.0f;
+    #if ENABLED(MPC_INCLUDE_FAN)
+      mpc.fan255_adjustment = 0.0f;
+    #endif
     mpc.block_heat_capacity = mpc.ambient_xfer_coeff_fan0 / block_responsiveness;
     mpc.sensor_responsiveness = block_responsiveness / (1.0f - (ambient_temp - asymp_temp) * exp(-block_responsiveness * t1_time) / (t1 - asymp_temp));
 

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -1085,11 +1085,9 @@ volatile bool Temperature::raw_temps_ready = false;
           block_responsiveness = -log((t2 - asymp_temp) / (t1 - asymp_temp)) / (sample_distance * (sample_count >> 1));
 
     mpc.ambient_xfer_coeff_fan0 = mpc.heater_power * (MPC_MAX) / 255 / (asymp_temp - ambient_temp);
-    #if ENABLED(MPC_INCLUDE_FAN)
-      mpc.fan255_adjustment = 0.0f;
-    #endif
     mpc.block_heat_capacity = mpc.ambient_xfer_coeff_fan0 / block_responsiveness;
     mpc.sensor_responsiveness = block_responsiveness / (1.0f - (ambient_temp - asymp_temp) * exp(-block_responsiveness * t1_time) / (t1 - asymp_temp));
+    TERN_(MPC_INCLUDE_FAN, mpc.fan255_adjustment = 0.0f);
 
     hotend.modeled_block_temp = asymp_temp + (ambient_temp - asymp_temp) * exp(-block_responsiveness * (ms - heat_start_time) / 1000.0f);
     hotend.modeled_sensor_temp = current_temp;

